From 012a41451cdfb7ba4419a54213d8d052d4423716 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 31 Jul 2026 15:20:53 +0200
Subject: [PATCH 20/27] playlist/podcast: fix duration parsing

itunes:duration can be HH:MM:SS, MM:SS or bare seconds
---
 modules/demux/playlist/podcast.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/demux/playlist/podcast.c b/modules/demux/playlist/podcast.c
index 5613156173..749687e4d0 100644
--- a/modules/demux/playlist/podcast.c
+++ b/modules/demux/playlist/podcast.c
@@ -418,6 +418,8 @@ static vlc_tick_t strTimeToMTime( const char *psz )
         return vlc_tick_from_sec( ( h*60 + m )*60 + s );
     case 2:
         return vlc_tick_from_sec( h*60 + m );
+    case 1:
+        return vlc_tick_from_sec( h );
     default:
         return INPUT_DURATION_UNSET;
     }
-- 
2.50.1 (Apple Git-155)

